index.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * This plugin allows us to create shared chunks which contain all module
+ * dependencies that don't already exist in an existing shared chunk.
+ *
+ * All dependencies required to load a shared chunk appear in the 'parents'
+ * array on each chunk.
+ *
+ * A chunk is webpack terminology for a set of modules (.js files).  Entry
+ * chunks are special chunks which initially contain the entry point (a .js
+ * file) and all files in its dependency tree.
+ *
+ * As optimization of chunks occur, modules are moved out of entry chunks and
+ * into new chunks that this plugin creates.  The plugin also updates
+ * dependencies between chunks (via the 'parents' property on chunks) based on
+ * whether a module already exists in shared chunk or not.
+ *
+ * Shared chunks are created in the order they are specified.
+ *
+ * A set of shared chunks can be specified in a webpack.config.js like so:
```

Comment:
Thank you for soldiering through this! I feel like the word "chunk" doesn't mean anything anymore, haha 😆

Reply:
I'll have to make some chocolate **chunk** cookies when this is all over.  😝